test/blackbox-tests/test-cases/pkg/implicit-dune-constraint.t

```diff
@@ -28,3 +29,39 @@ constraint.
   $ test "4.0.0"
   Solution for dune.lock:
   - foo.0.0.1
+
+Create a fake project and ensure `dune` can be used as a dependency:
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name bar)
+  >  (allow_empty)
+  >  (depends dune))
+  > EOF
+  $ dune pkg lock
+  Internal error, please report upstream including the contents of _build/log.
+  Description:
+    ("can't find a valid solution for the dependencies", { name = "bar" })
+  Raised at Stdune__Code_error.raise in file
+    "otherlibs/stdune/src/code_error.ml", line 10, characters 30-62
+  Called from Dune_pkg__Opam_solver.reject_unreachable_packages.(fun) in file
+    "src/dune_pkg/opam_solver.ml", lines 830-832, characters 14-56
+  Called from Stdlib__Map.Make.merge in file "map.ml", line 406, characters
+    44-63
+  Called from Dune_pkg__Opam_solver.reject_unreachable_packages.(fun) in file
+    "src/dune_pkg/opam_solver.ml", lines 802-841, characters 6-32
+  Called from Dune_pkg__Opam_solver.solve_lock_dir.(fun) in file
+    "src/dune_pkg/opam_solver.ml", line 949, characters 10-78
+  Called from Fiber__Scheduler.exec in file "vendor/fiber/src/scheduler.ml",
+    line 76, characters 8-11
+  Re-raised at Stdune__Exn.raise_with_backtrace in file
+    "otherlibs/stdune/src/exn.ml", line 38, characters 27-56
+  Called from Fiber__Scheduler.exec in file "vendor/fiber/src/scheduler.ml",
+    line 76, characters 8-11
+  
+  I must not crash.  Uncertainty is the mind-killer. Exceptions are the
+  little-death that brings total obliteration.  I will fully express my cases. 
+  Execution will pass over me and through me.  And when it has gone past, I
+  will unwind the stack along its path.  Where the cases are handled there will
+  be nothing.  Only I will remain.
+  [1]
```

Comment:
Can you adjust the test to filter out just e.g. the description? The whole stacktrace is inconvenient as the line numbers will cause unrelated failures.